fdbserver/DataDistribution.actor.cpp

```diff
@@ -3119,7 +3168,11 @@ ACTOR Future<Void> serverMetricsPolling( TCServerInfo *server) {
 //Returns the KeyValueStoreType of server if it is different from self->storeType
 ACTOR Future<KeyValueStoreType> keyValueStoreTypeTracker(DDTeamCollection* self, TCServerInfo *server) {
 	state KeyValueStoreType type = wait(brokenPromiseToNever(server->lastKnownInterface.getKeyValueStoreType.getReplyWithTaskID<KeyValueStoreType>(TaskPriority::DataDistribution)));
-	if(type == self->configuration.storageServerStoreType && (self->includedDCs.empty() || std::find(self->includedDCs.begin(), self->includedDCs.end(), server->lastKnownInterface.locality.dcId()) != self->includedDCs.end()) )
+	if (type == self->configuration.storageServerStoreType &&
+	    (self->includedDCs.empty() ||
+	     std::find(self->includedDCs.begin(), self->includedDCs.end(), server->lastKnownInterface.locality.dcId()) !=
+	         self->includedDCs.end()) &&
+	    (self->isValidLocality(self->configuration.storagePolicy, server->lastKnownInterface.locality)))
```

Comment:
I think this could also use braces for clarity now that the `if` condition is a complex multi-line expression.